### DISCUSSION TITLE
Unable to deploy server with https

### DISCUSSION THREAD

**Discussion by SlitiBrahim** [pocketbase/pocketbase#3179] [Q&A]
Hey!

I am trying to deploy my pocketbase backend on an ec2 instance, it works really fine when I use `serve http=0.0.0.0:80` and I can access it from the internet. But when I use the domain that points to my server (100% sure the domain redirection works: using `serve http=app.mydomain.com:80` works totally fine when accessing `app.mydomain.com` in a browser) by running `serve --http=app.mydomain.com:80` or `serve --https=app.mydomain.com` I get the error `2023/08/23 22:32:55 listen tcp x.x.x.x:443: bind: cannot assign requested address`, where `x.x.x.x` is the IP addr of my server. I've double checked and those ports are not used. Please note that I run the command with `sudo`.

I can confirm my domain (`app.mydomain.com`, example here oc) points to this server.
Also, not necessarily related as it doesn't work on the linux machine (Ubuntu 22.04) itself but I have an external firewall (security group) that allows https and http inbound connections.

Is there any step I've missed? 

Thanks a lot for your work, I was able to build an MVP really fast and I could never have been finished it that fast without your work! ❤️

**Comment by azizvc** [comment]
Look this https://pocketbase.io/docs/going-to-production/#using-reverse-proxy

**Reply by azizvc** [reply]
Just to complement, when you are deploying on Ec2 you need to:

-  install a webserver like apache or nginx,
- configure webserver to do a reverse proxy from 443 to localhost:80 
- setup the ssl certificate (use certbot)

too complicated? use this: [caddyserver](https://caddyserver.com/docs/quick-starts/reverse-proxy) 
is easier than apache and nginx

**Reply by ganigeorgiev** [reply]
_Just for info since based on the below [author comment](https://github.com/pocketbase/pocketbase/discussions/3179#discussioncomment-6811601) this has caused some confusion - using a reverse proxy or certbot is not a requirement to use PocketBase. If you don't need the reverse proxy capabilities, you should be able to start a web server with Let's encrypt certificate by just using the executable._

**Comment by ganigeorgiev** [comment]
> But when I use the domain that points to my server 

But how did you setup that part?

When you use the flags `--http=domain.com:80` or `--https=domain.com:443`, the Go dns resolver tries to bind only to the IP directly resolving from "domain.com", meaning that you most likely miss an entry in your `/etc/hosts` file that maps the public ip/domain and your local vm. This usually is done automatically by most VPS providers when using their domain management tools, but I'm not sure if that't the case with AWS. For an example, see the users suggestions in https://github.com/pocketbase/pocketbase/issues/107#issuecomment-1331714755.

_This eventually could be avoided if a version of https://github.com/pocketbase/pocketbase/pull/2795 is implemented, because this way we can allow both binding to all IPv4/IPv6 local interfaces (`0.0.0.0`/`[::]`) and specify the TLS domain(s)._

**Comment by SlitiBrahim** [comment]
Hey!

Oh damn I just thought pocketbase was embedding a certbot, I think I heard this somewhere so indeed I didn't setup any reverse proxy! I will setup an nginx with a certbot and see how it goes.

Thank you

**Reply by ganigeorgiev** [reply]
PocketBase doesn't embed cerbot and it doesn't need it (the same apply for a reverse proxy).
The Let's encrypt certificate is auto issued and managed by the standard Go [`crypto/tls` package](https://pkg.go.dev/crypto/tls).
Please check https://github.com/pocketbase/pocketbase/discussions/3179#discussioncomment-6808938.
